feat(ec2): add `versionDescription` property for `LaunchTemplate`

yuppe-kyupeen opened the pull request:
### Issue # (if applicable)

### Reason for this change
The change introduces the `versionDescription` property to the `LaunchTemplate`

### Description of changes
- Add the `versionDescription` property for `LaunchTemplateProps`, which was missing in the L2 construct.
- Add validation for character limit 

### Description of how you validated changes
I Added a unit test for launch template and added the `versionDescription` property in the integration tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: c3d69db312d6187c52487a17d9d3b4eef84db38f
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ca888cb2-9e96-4028-a8ff-b1e4c15ddc8c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented on packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:
```diff
@@ -221,6 +221,15 @@ export interface LaunchTemplateProps {
    */
   readonly launchTemplateName?: string;
 
+  /**
+   * A description for the first version of the launch template.
+   * The version descrioption must be maximum 255 characters long, including hyphens (-),
+   * underscores (_), spaces, and tabs.
+   *
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-launchtemplate.html#cfn-ec2-launchtemplate-versiondescription
+   */
+  readonly versionDescription?: string;
```
The CodeBuild CI seems to be failing. Optional properties must have `@default` documentation. Could you add the label and description?

```
aws-cdk-lib: error: [awslint:props-default-doc:aws-cdk-lib.aws_ec2.LaunchTemplateProps.versionDescription] Optional property must have @default documentation 
aws-cdk-lib: Error: /codebuild/output/src1195990305/src/github.com/aws/aws-cdk/tools/@aws-cdk/cdk-build-tools/bin/cdk-awslint exited with error code 1
aws-cdk-lib: Build failed.!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

go-to-k commented on packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:
```diff
@@ -221,6 +221,15 @@ export interface LaunchTemplateProps {
 
+  /**
+   * A description for the first version of the launch template.
+   * The version descrioption must be maximum 255 characters long, including hyphens (-),
+   * underscores (_), spaces, and tabs.
```
`including hyphens (-),`
`   * underscores (_), spaces, and tabs.`

Is there documentation available to confirm this? (I could not find that in the CFn documentation.)

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-launchtemplate.html#cfn-ec2-launchtemplate-versiondescription

go-to-k commented on packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:
```diff
@@ -221,6 +221,15 @@ export interface LaunchTemplateProps {
   readonly launchTemplateName?: string;
 
+  /**
+   * A description for the first version of the launch template.
+   * The version descrioption must be maximum 255 characters long, including hyphens (-),
```
A line break would make it clearer.

```suggestion
   * A description for the first version of the launch template.
   * 
   * The version descrioption must be maximum 255 characters long, including hyphens (-),
```

yuppe-kyupeen commented:
@go-to-k 
Thank you for the review!
I have addressed the points you mentioned.I apologize for any oversights.
Please review the changes.Thank you!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 71633857a8f7e1acdd03886d48d013a169ad2274
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5f2d67f7-35b5-4029-97f2-3ecc2217f7ef%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 849e005250e39e2497dc41f74b57f2335c444733
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cfb68953-0546-4df6-8fc4-e6864a035444%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented on packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:
```diff
@@ -221,6 +221,17 @@ export interface LaunchTemplateProps {
+   *
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-launchtemplate.html#cfn-ec2-launchtemplate-versiondescription
+   *
+   * @default - The version description is not set if not specified.
```
I think it could be simpler.

```suggestion
   * @default - no description
```

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0bec9e2c268f61dabe8e6ddf9f0ab7f78b6dfa04
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bb2eabf2-a2dc-4647-a4e1-9052cecbe104%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k reviewed (approved):
Thanks for the changes. Approved.

yuppe-kyupeen commented:
@go-to-k 
thank you!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 115a68c2f4a9b107d3c29a7512668819eaee25f2
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=34cc6303-3fbf-4d00-8034-7f3f440229bd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

gracelu0 commented on packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts:
```diff
@@ -221,6 +221,17 @@ export interface LaunchTemplateProps {
+  /**
+   * A description for the first version of the launch template.
+   *
+   * The version descrioption must be maximum 255 characters long.
```
```suggestion
   * The version description must be maximum 255 characters long.
```

gracelu0 reviewed (commented):
Hi @yuppe-kyupeen , thanks for your contribution. Just one small typo to fix, otherwise LGTM!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 4a1505ba5e808a7c249cc5e92e202f048d3c8d95
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=54b66771-937a-4022-83f5-30b4ded232d4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: a2798a5eee5436ad2f39bc08e6cc617a9046fe19
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a80f2022-ce10-49e7-abcf-1d0556f52aeb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

yuppe-kyupeen commented:
Hi @gracelu0, thanks for your review and suggestion! 👍

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 70c9be5ad62fa79b7f4e0998ca8b2c279b3b1d83
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=be00a5fb-d47f-49a1-a20a-1b3295bb9368%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
Comments on closed issues and PRs are hard for our team to see. If you need help, please open a new issue that references this one.
